src/RwaUrnUtils.t.sol

```diff
@@ -0,0 +1,287 @@
+        RwaOutputConduit2(mate).push();
+
+        // Fast-forward 30 day
+        hevm.warp(block.timestamp + 30 days);
+        // Mints some additional Dai into the receiver's balance to cope with accrued fees
+        mintDai(rec, 10 ether);
+
+        // Makes sure daiJoin has enought Dai available
+        mintDai(address(daiJoin), 100 ether);
+
+        uint256 expectedAmount = urnUtils.estimateWipeAllWad(address(urn), block.timestamp);
+        uint256 recBalanceBefore = dai.balanceOf(rec);
+
+        urnUtils.pushAndWipeAll(address(urn), address(inConduit), rec);
+
+        uint256 recBalanceAfter = dai.balanceOf(rec);
+
+        // Get the remaining debt in the urn
+        (, uint256 art) = vat.urns("RWA008AT1-A", address(urn));
+        // If the value is >0, it must revert
+        assertEq(art, 0);
+        assertEq(recBalanceBefore - recBalanceAfter, expectedAmount);
+    }
+
+    function testFullRepaymentWhenUrnHasOutstandingDai() public {
+        ForwardProxy(op).updateForwardTo(address(urn));
+        RwaUrn2(op).lock(1 ether);
+        RwaUrn2(op).draw(400 ether);
+
+        ForwardProxy(op).updateForwardTo(address(outConduit));
+        RwaOutputConduit2(op).pick(rec);
+
+        ForwardProxy(mate).updateForwardTo(address(outConduit));
+        RwaOutputConduit2(mate).push();
+
+        // Fast-forward 30 day
+        hevm.warp(block.timestamp + 30 days);
+
+        // Mints some additional Dai into the receiver's balance to cope with accrued fees
+        mintDai(rec, 10 ether);
+        // Mints some additional Dai into the urn
+        mintDai(address(urn), 50 ether);
+
+        // Makes sure daiJoin has enought Dai available
+        mintDai(address(daiJoin), 100 ether);
+
+        uint256 recBalanceBefore = dai.balanceOf(rec);
+        uint256 urnBalanceBefore = dai.balanceOf(address(urn));
+        uint256 expectedAmount = urnUtils.estimateWipeAllWad(address(urn), block.timestamp);
+
+        urnUtils.pushAndWipeAll(address(urn), address(inConduit), rec);
+
+        uint256 recBalanceAfter = dai.balanceOf(rec);
+
+        assertEq(recBalanceBefore - recBalanceAfter, expectedAmount - urnBalanceBefore);
+    }
+
+    function testFailFullRepaymentWhenPayerHasNotEnoughDai() public {
+        ForwardProxy(op).updateForwardTo(address(urn));
+        RwaUrn2(op).lock(1 ether);
+        RwaUrn2(op).draw(400 ether);
+
+        ForwardProxy(op).updateForwardTo(address(outConduit));
+        RwaOutputConduit2(op).pick(rec);
+
+        ForwardProxy(mate).updateForwardTo(address(outConduit));
+        RwaOutputConduit2(mate).push();
+
+        // Fast-forward 30 day
+        hevm.warp(block.timestamp + 30 days);
+
+        // Makes sure daiJoin has enought Dai available
+        mintDai(address(daiJoin), rad(10 ether));
+
+        urnUtils.pushAndWipeAll(address(urn), address(inConduit), rec);
+    }
+
+    function mintDai(address who, uint256 wad) internal {
+        dai.mint(who, wad);
+        vat.suck(VOW, who, rad(wad));
```

Comment:
Suck mints Dai as well, you are minting 2x wad Dai here. What's the reasoning for this? Why even have the suck step, if you can just mint Dai directly?